src/components/Menu/MenuItem.tsx

```diff
@@ -179,6 +297,44 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
 
     _.invoke(this.props, 'onFocus', e, this.props)
   }
+
+  private closeMenu = e => {
+    const { menu, inSubmenu } = this.props
+    const { submenuOpen } = this.state
+    if (menu && submenuOpen) {
+      this.setState({ submenuOpen: false })
+      if (!inSubmenu && (keyboardKey.getCode(e) === keyboardKey.Escape || this.props.vertical)) {
```

Comment:
what is a use case to have an additional check on ```Esc``` button? We should avoid using keys in components, if you need a specific thing to happen on Esc, better create a separate action handler for that.

Reply:
Added new handler, thanks for the suggestion!